Entities/Industry/Fireplace/Fireplace.as

```diff
@@ -48,10 +48,38 @@ void onCollision(CBlob@ this, CBlob@ blob, bool solid)
 {
 	if (blob !is null)
 	{
+		// Convert items (and animals) into food blobs (different textures and names from your regular !food)
 		if (blob.getName() == "fishy" && this.getSprite().isAnimation("fire"))
 		{
 			blob.getSprite().PlaySound("SparkleShort.ogg");
-			CBlob@ food = server_MakeFood(blob.getPosition(), "Cooked Fish", 1);
+			CBlob@ food = server_MakeFood(blob.getPosition(), "Cooked Fish", 1); // Cooked Fish
```

Comment:
I feel like the repetition here could be reduced quite a lot. Consider checking the right animation ahead of all the name checks, a `bool cooked` above all of this, and killing the blob, setting the food velocity, and playing the sound if it _was_ cooked. Should cut the amount of code in half and make it much more future proof.

Reply:
(to be clear, the line highlighted is just the start of a big highly repetitive section of code, not a particularly problematic line)